tests/zquantum/qcbm/ansatz_test.py

```diff
@@ -444,53 +406,44 @@ def test_ansatz_circuit_eight_layers(self, number_of_qubits, topology):
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[0][i + number_of_qubits], i)
-            )
-        expected_first_layer = Circuit(expected_pycircuit)
+            expected_first_layer += RZ(params[0][i + number_of_qubits])(i)
+
         expected_second_layer = get_entangling_layer(
-            params[1], number_of_qubits, "XX", topology
+            params[1], number_of_qubits, XX, topology
         )
-        expected_pycircuit = Program()
+
+        expected_third_layer = Circuit()
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(pyquil.gates.RX(params[2][i], i))
+            expected_third_layer += RX(params[2][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[2][i + number_of_qubits], i)
-            )
-        expected_third_layer = Circuit(expected_pycircuit)
+            expected_third_layer += RZ(params[2][i])(i)
+
         expected_fourth_layer = get_entangling_layer(
-            params[3], number_of_qubits, "XX", topology
+            params[3], number_of_qubits, XX, topology
         )
-        expected_pycircuit = Program()
+        expected_fifth_layer = Circuit()
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(pyquil.gates.RX(params[4][i], i))
+            expected_fifth_layer += RX(params[4][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[4][i + number_of_qubits], i)
-            )
-        expected_fifth_layer = Circuit(expected_pycircuit)
+            expected_fifth_layer += RZ(params[4][i + number_of_qubits])(i)
+
         expected_sixth_layer = get_entangling_layer(
-            params[5], number_of_qubits, "XX", topology
+            params[5], number_of_qubits, XX, topology
         )
-        expected_pycircuit = Program()
+        expected_seventh_layer = Circuit()
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(pyquil.gates.RX(params[6][i], i))
+            expected_seventh_layer += RX(params[6][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[6][i + number_of_qubits], i)
-            )
+            expected_seventh_layer += RZ(params[6][i + number_of_qubits])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RX(params[6][i + 2 * number_of_qubits], i)
-            )
-        expected_seventh_layer = Circuit(expected_pycircuit)
+            expected_seventh_layer += RX(params[6][i + 2 * number_of_qubits])(i)
+
         expected_eigth_layer = get_entangling_layer(
-            params[7], number_of_qubits, "XX", topology
+            params[7], number_of_qubits, XX, topology
```

Comment:
WDYT about using a list comprehension to define the expected circuit at once? It would look like this:
```python
        n_qubits = number_of_qubits  # make the variable name shorter for better noise v. signal ratio
        expected_circuit = Circuit(
            [
                # first layer
                *[RX(params[0][i])(i) for i in range(n_qubits)],
                *[RZ(params[0][i + n_qubits])(i) for i in range(n_qubits)],
                # second layer
                get_entangling_layer(params[1], n_qubits, XX, topology),
                # third layer
                *[RX(params[2][i])(i) for i in range(n_qubits)],
                *[RZ(params[2][i])(i) for i in range(n_qubits)],
                # fourth layer
                get_entangling_layer(params[3], n_qubits, XX, topology),
                # fifth layer
                *[RX(params[4][i])(i) for i in range(n_qubits)],
                *[RZ(params[4][i + n_qubits])(i) for i in range(n_qubits)],
                # sixth layer
                get_entangling_layer(params[5], n_qubits, XX, topology),
                # seventh layer
                *[RX(params[6][i])(i) for i in range(n_qubits)],
                *[RZ(params[6][i + n_qubits])(i) for i in range(n_qubits)],
                *[RZ(params[6][i + 2 * n_qubits])(i) for i in range(n_qubits)],
                # eight layer
                get_entangling_layer(params[7], number_of_qubits, XX, topology)
            ]
        )
```

Using for loops vs expressions and comprehensions is subject to personal preference, but in this use case it might be more than that. It's a little bit easier to catch bugs because the whole circuit definition fits entirely on a single screen, so we can see any anomalies in the code _shape_. For example, the third layer looks odd to me and indeed this might be a bug @dexter2206 mentioned in another comment.

Reply:
Done in https://github.com/zapatacomputing/z-quantum-qcbm/pull/40/commits/60f9b2a2d6fba739091b7c3bfa242c8f77338b74